File: LeetCode/JavaScript/1456 - Maximum Number of Vowels in a Substring of Given Length.js
/**
1456. Maximum Number of Vowels in a Substring of Given Length
https://leetcode.com/problems/maximum-number-of-vowels-in-a-substring-of-given-length/
*/

/**
 * @param {string} s
 * @param {number} k
 * @return {number}
 */
var maxVowels = function(s, k) {

    var vowels = ['a', 'e', 'i', 'o', 'u'];
    var count = 0;
    var temp = 0;
    var i = 0;
    while (i <= s.length - 1)

    {
        if (vowels.includes(s[i]))
            temp += 1;

        if (i >= k && vowels.includes(s[i - k]))
            temp--;

        count = Math.max(temp, count);

        i += 1;
    }

    return count;
};

File: LeetCode/Python/1456 - Maximum Number of Vowels in a Substring of Given Length.py
'''
1456. Maximum Number of Vowels in a Substring of Given Length
https://leetcode.com/problems/maximum-number-of-vowels-in-a-substring-of-given-length/
'''


def maxVowels(self, s: str, k: int) -> int:

    vowels = ['a', 'e', 'i', 'o', 'u']
    vowel_count = 0

    counter = 0

    for j in range(len(s)):

        if j >= k:
            if s[j-k] in vowels:
                counter -= 1

        if s[j] in vowels:
            counter += 1

        vowel_count = max(counter, vowel_count)

    return vowel_count
